src/main/java/org/cloudfoundry/autosleep/dao/model/AutosleepServiceInstance.java

```diff
@@ -62,6 +62,8 @@ private AutosleepServiceInstance() {
 
     public AutosleepServiceInstance(CreateServiceInstanceRequest request) throws HttpMessageNotReadableException {
         super(request);
+        //only set secret on creation
+        setSecretFromParams(request.getParameters());
```

Comment:
This param is not mandatory on creation. It's mandatory whenever we want to set a protected parameter. Exemple: create service without no_optout (and no secret). Update service with no_optout and secret